openest/test/generate/test_functions.py

```diff
@@ -1,66 +1,69 @@
-import datetime
 import numpy as np
+import pandas as pd
 import xarray as xr
+
 from openest.generate.base import Constant
 from openest.generate.daily import YearlyDayBins
 from openest.generate.functions import Scale, Instabase, SpanInstabase
 from test_daily import test_model
 
-def make_year_ds(year, values):
-    return xr.Dataset({'x': (['time'], values)},
-                      coords={'time': map(lambda nn: datetime.date(year, 1, 1) + datetime.timedelta(nn), np.arange(365))})
 
-def test_Scale():
+def make_year_da(year, values):
+    year = str(year)
+    time_coord = pd.date_range(year + '-01-01', periods=365)
+    return xr.DataArray(values, coords=[('time', time_coord)])
+
+
+def test_scale():
     application = Scale(Constant(33, 'widgets'), {'test': 1. / 11}, 'widgets', 'subwigs').test()
-    for yearresult in application.push(make_year_ds(2000, np.random.rand(365))):
+    for yearresult in application.push(make_year_da(2000, np.random.rand(365))):
         np.testing.assert_equal(yearresult[1], 3)
 
     def check_units():
-        application = Scale(Constant(33, 'widgets'), {'test': 1.8}, 'deg. C', 'deg F.').test()
-        for yearresult in application.push(make_year_ds(2000, np.random.rand(365))):
+        app = Scale(Constant(33, 'widgets'), {'test': 1.8}, 'deg. C', 'deg F.').test()
+        for _ in app.push(make_year_da(2000, np.random.rand(365))):
             return
 
     np.testing.assert_raises(Exception, check_units)
 
-def test_Instabase():
+
+def test_instabase():
     calc = Instabase(YearlyDayBins(test_model, 'days'), 2)
     app1 = calc.test()
     app2 = calc.test()
-    for yearresult in app1.push(make_year_ds(1800, [-10] * 300 + [10] * 65)):
-        np.testing.assert_equal(True, False) # Should get nothing here
-    for yearresult in app2.push(make_year_ds(1800, [10] * 365)):
-        np.testing.assert_equal(True, False) # Should get nothing here
-    for yearresult in app1.push(make_year_ds(2000, [-10] * 65 + [10] * 300)):
+    for _ in app1.push(make_year_da(1800, [-10] * 300 + [10] * 65)):
+        raise AssertionError  # Should get nothing here
+    for _ in app2.push(make_year_da(1800, [10] * 365)):
+        raise AssertionError  # Should get nothing here
```

Comment:
I thought that `np.testing`'s `assert` functions were aware of testing environments and would not actually break the test suite. If not, is there a way to have this functionality?

Reply:
You can still use that numpy.testing but numpy.testing isn't really a part of this. This is all `pytest`. I think that `raise` the error we're making it very explicit that we wanted the error and its not a typo.